### PR TITLE
Adding in new isDate function

### PR DESCRIPTION
**Pull request by dalefrancis88**
Adding in the new isDate predicate as suggested by @karthikiyengar 

A question i'd like to raise is whether we want to have invalid dates return as a true or false. My argument is that for the 99% of use cases people are going to want to know it's a valid date and move on. It appears this is the same conclusion we've come too in `isNumber` as well.

### CODE REVIEW

File: src/core/isDate.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Dale Francis (dalefrancis88) */
+
+const hasAlg = require('./hasAlg')
+
+// isDate : a -> Boolean
+function isDate(x) {
+  return hasAlg('valueOf', x) && x instanceof Date && !isNaN(x.valueOf())
```
[the remark this answers is not in the document]
Comment (dalefrancis88) [reply src/core/isDate.js]: fair enough good to know :) are you suggesting to change to the first or wrap `isSameType`?

[the remark this answers is not in the document]
Comment (evilsoft) [reply src/core/isDate.js]: I would go with the first one

[the remark this answers is not in the document]
Comment (dalefrancis88) [reply src/core/isDate.js]: Yeah that's what i did, should have come back and commented 😄

**Comment by evilsoft** [comment]
:eyes:

**Comment by dalefrancis88** [comment]
Did you mean to approve?

File: src/core/inspect.js [hunk 2]
```diff
@@ -37,7 +38,7 @@ function inspect(x) {
     return ` "${x}"`
   }
 
-  if(isSymbol(x)) {
+  if(isSymbol(x) || isDate(x)) {
```
Comment (evilsoft) [inline src/core/inspect.js]: Curious why we do this?
In the browser should output the string version of the date already.
Is there a case you know of where this is not true?

Comment (dalefrancis88) [reply src/core/inspect.js]: So doing no conversion does returns something like this, "2018-12-23T01:04:17.707Z" whereas calling `toString` returns this, "Sat Dec 22 2018 17:07:07 GMT-0800 (Pacific Standard Time)" which I thought was a nicer version to see for inspect, what do you think?

**Pull request merged**